### ISSUE TITLE
Question: scaling application container risk ?

### ISSUE THREAD

**Issue by doc75** [nextcloud/docker#859]
Hello,

I’am using nextcloud container alpine-fpm, and I am investigating the ability to scale this container using docker-compose.

Is there any risk to start several nextcloud container at the same time sharing the same directory and the same DB (with a redis container for memcache) ?
My question is especially related to the upgrade process launched  when changing version ? Can’t we have several migration launched in parallel in each of the nextcloud container ?

Thanks in advance for your answer.

**Comment by J0WI** [comment]
Make sure that you use Redis for file locking and that you share this container for all FPM containers.
If you also share the same volume(s) for all containers the upgrade is the same as with just one FPM node. Major upgrades and database changes will still cause a downtime.

You can find some further recommendations here:
https://docs.nextcloud.com/server/11/admin_manual/installation/deployment_recommendations.html
The Lookup-Server architecture scales even better:
https://github.com/nextcloud/lookup-server/blob/master/doc/architecture.md

**Comment by doc75** [comment]
@J0WI, thanks for your answer.
I am indeed afraid that 2 nextcloud containers sharing the same DB and Volumes will try to trigger the upgrade procedure at the same time during startup. It is not something that need to worry me ?

**Comment by J0WI** [comment]
We rely on `rsync` and `occ` for the update process. I haven't had any issues with them.

**Comment by SimJoSt** [comment]
I have successfully connected redis to my setup and on container start it tells me the distributed session storage is setup via redis.  
If I use the `docker-compose up -d --scale nextcloud=2` with 2 or more, I get issues in the frontend. 404s, information that have no rights to access that resource, etc.

Are there any more steps necessary to make this run smoothly? I'll gladly open a new issue, but this one seemed to fit the scope, as apparently there is a risk :)

**Comment by SimJoSt** [comment]
Never mind. My issue was, that I didn't have the entire `/var/www/html` in a volume and didn't understand how the container checks at startup up, if an update of the source files should be performed etc.  
After fixing that, it worked like a breeze.
